sql/flaky-test-ratio.sql

```diff
@@ -0,0 +1,14 @@
+SELECT format_datetime(runs.created_at, 'Y:w') year_week
+     , jobs.run_id
+     , runs.created_at
+     , jobs.name
+     , count(*) FILTER (WHERE (jobs.conclusion = 'success')) success_count
+, count(*) FILTER (WHERE (jobs.conclusion = 'failure')) failure_count
+, round(((CAST(count(*) FILTER (WHERE (jobs.conclusion = 'failure')) AS double) / CAST((count(*) FILTER (WHERE (jobs.conclusion = 'success')) + count(*) FILTER (WHERE (jobs.conclusion = 'failure'))) AS double)) * 100), 2) flaky_ratio_percent
+FROM jobs
+         JOIN runs ON jobs.run_id = runs.id
+WHERE NOT jobs.name LIKE 'Test Report%'
+GROUP BY 1, jobs.run_id, runs.created_at, jobs.name
+HAVING ((count(*) FILTER (WHERE (jobs.conclusion = 'success')) > 0) AND (count(*) FILTER (WHERE (jobs.conclusion = 'failure')) > 0))
+ORDER BY year_week, 6 DESC, jobs.name, jobs.run_id
```

Comment:
If you'd want to avoid repeating these `count` expressions, you could put this query with `success_count` and `failure_count` in a CTE, and calculate this ratio in the main query.
```suggestion
WITH jobs_with_multiple_attempts AS (
  SELECT
    run_id
    , name
    , count(*) FILTER (WHERE (conclusion = 'success')) success_count
    , count(*) FILTER (WHERE (conclusion = 'failure')) failure_count
  FROM jobs
  WHERE name NOT LIKE 'Test Report%'
  GROUP BY 1, 2
  HAVING count(*) > 1
)
SELECT
    format_datetime(runs.created_at, 'Y:w') year_week
    , jobs.run_id
    , runs.created_at
    , jobs.name
    , jobs.success_count
    , jobs.failure_count
    , round(((CAST(jobs.failure_count AS double) / CAST((jobs.success_count + jobs.failure_count) AS double)) * 100), 2) flaky_ratio_percent
FROM jobs_with_multiple_attempts jobs
         JOIN runs ON jobs.run_id = runs.id
GROUP BY 1, 2, 3, 4
ORDER BY 1, 6 DESC, 4, 2
```

Reply:
Some more polishing done - hope looks better